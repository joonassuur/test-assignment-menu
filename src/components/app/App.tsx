import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { getDropdown, setDropdown } from "../../redux/Index";
import Header from "../header/Header";
import DropdownMain from "../dropdown-main/DropdownMain";
import "./App.scss";

function App() {
  const dropdown = useSelector(getDropdown);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(setDropdown(false));
  }, [dispatch]);

  return (
    <div className="App">
      <Header></Header>
      {dropdown && <DropdownMain />}
    </div>
  );
}

export default App;
