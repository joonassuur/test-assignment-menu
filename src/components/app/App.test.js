import React from "react";
import TestRenderer from "react-test-renderer";
import App from "./App";

const mockDispatch = jest.fn();

jest.mock("react-redux", () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../redux/Index");

describe("App component", () => {
  test("Matches the snapshot", () => {
    const app = TestRenderer.create(<App />);
    expect(app.toJSON()).toMatchSnapshot();
  });
});
