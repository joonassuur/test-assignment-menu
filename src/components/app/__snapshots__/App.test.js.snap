// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="header-container"
  >
    <div
      className="user-info"
    >
      <div
        className="title"
      >
        Elon Musk
      </div>
      <div
        className="subtitle"
      />
    </div>
    <div
      className="settings-icon"
      onClick={[Function]}
    >
      <span
        className="material-icons-outlined"
      >
        settings
      </span>
    </div>
  </div>
</div>
`;
